### PR TITLE
Create codeql-analysis.yml

### PR DESCRIPTION
**Pull request by normanmaurer**
Motivation:

Github now allows to run CodeQL during pull request verification. This allows to detect errors / security problems early.

Modification:

Add config

Result:

Fixes https://github.com/netty/netty/issues/10669

### CODE REVIEW

**Review by hyperxpro — APPROVED** [review]
Finally. :)

**Comment by chrisvest** [comment]
Nice.

File: .github/workflows/codeql-analysis.yml [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+  analyze:
+    name: Analyze
+    runs-on: ubuntu-latest
+
+    strategy:
+      fail-fast: false
+      matrix:
+        # Override automatic language detection by changing the below list
+        # Supported options are ['csharp', 'cpp', 'go', 'java', 'javascript', 'python']
+        language: ['java']
```
Comment (artem-smotrakov) [inline .github/workflows/codeql-analysis.yml:27]: The projects has some JNI parts written in C. I am wondering in we can also scan C code with CodeQL.

```suggestion
        language: ['java', 'cpp' ]
```

Not sure that `c` can be used in the `language` filed. I didn't find it in the docs.

Comment (normanmaurer) [reply .github/workflows/codeql-analysis.yml:27]: Seems like c is not supported atm

Comment (artem-smotrakov) [reply .github/workflows/codeql-analysis.yml:27]: The docs say that both C/C++ are supported. I think we can use `cpp` to enable scans for the C code.

https://docs.github.com/en/free-pro-team@latest/github/finding-security-vulnerabilities-and-errors-in-your-code/about-code-scanning

Comment (normanmaurer) [reply .github/workflows/codeql-analysis.yml]: seems like there needs to be some more setup to make this work :/

Comment (artem-smotrakov) [reply .github/workflows/codeql-analysis.yml]: Yeah, it looks so. Maybe we can disable autobuild and add the following

```
- run: mvn -DskipTests install
```

Or, if it doesn't work, maybe add `if` blocks and build each JNI module separately with autogen.sh + make.

Anyway, I think we can add it later.

Comment (normanmaurer) [reply .github/workflows/codeql-analysis.yml]: let me try...

Comment (normanmaurer) [reply .github/workflows/codeql-analysis.yml]: @artem-m seems like this works ? :)

Comment (artem-smotrakov) [reply .github/workflows/codeql-analysis.yml]: Yeah, looks like it works! :) I don't see the results at the moment in the tab "Code scanning results". But maybe they're going to be available later (it says "This usually happens when you enable code scanning for the first time").

**Review by artem-smotrakov — APPROVED** [review]

**Pull request merged**